spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -48,49 +51,55 @@
 	public static final int DEFAULT_EXECUTOR_THREADS = 4;
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForPublisher() {
+	public ExecutorProvider publisherExecutorProvider() {
```

Comment:
IIUC, we can't use `@ConditionalOnMissingBean` here because one of these `ExecutorProviders` gets created first, and the other one never gets created. I think I ran into this issue a few days ago. I don't think `@ConditionalOnMissingBean` takes into account the bean's qualified name, maybe you can confirm?